packages/consumption/src/modules/requests/itemProcessors/shareAttribute/ShareAttributeRequestItemProcessor.ts

```diff
@@ -130,6 +130,18 @@ export class ShareAttributeRequestItemProcessor extends GenericRequestItemProces
             if (nonPendingRelationshipsToPeer.length === 0) {
                 return ValidationResult.error(ConsumptionCoreErrors.requests.cannotShareRelationshipAttributeOfPendingRelationship());
             }
+
+            if (!requestItem.thirdPartyAddress) {
+                return ValidationResult.error(
+                    ConsumptionCoreErrors.requests.invalidRequestItem(
+                        "The source attribute provided is a relationship attribute. You must provide a third party address that is the original peer when sharing with a third party."
+                    )
+                );
+            }
+
+            if (!requestItem.thirdPartyAddress.equals(foundAttribute.shareInfo.peer)) {
+                return ValidationResult.error(ConsumptionCoreErrors.requests.invalidRequestItem("The third party address must be the peer of the relationship attribute."));
+            }
```

Comment:
I think it would be better to write RelationshipAttribute as in the other Errormessages above. And `You must provide a third party address that is the original peer when sharing with a third party.` is perhaps not easy to understand. What do you think of 
`You must specify the peer as thirdPartyAddress with whom you shared originally this attribute.`

Reply:
Yea the thing is that the original is not only the one you shared with someone but also one that someone shared with you